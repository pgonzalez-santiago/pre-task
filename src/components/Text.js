/* Global imports */
import React from 'react'
import { string, number, oneOfType, bool } from 'prop-types'
import { connect } from 'react-redux'

/* Local imports */
import I18n from '../assets/lang'
import { getCurrent } from '../store/reducers/language'

/* Component definition */
export const Text = ({ currentLang, className, translate, children: value }) => {
  return (
    <span className={className} key={currentLang}>
      { getValue(translate, value) }
    </span>
  )
}
/* PropTypes */
Text.propTypes = {
  children:    oneOfType([string, number]),
  className:   string,
  currentLang: string.isRequired,
  translate:   bool,
}

Text.defaultProps = {
  defaultValue: '[...]',
  translate:    true,
}

/* Local utility functions */
export const getValue = (shouldTranslate, value) => {
  return shouldTranslate ? I18n.t(value) : value
}

/* Styles */

const mapStateToProps = state => ({
  currentLang: getCurrent(state),
})

export default connect(mapStateToProps)(Text)
