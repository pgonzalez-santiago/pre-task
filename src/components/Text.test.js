import React from 'react'
import { Text } from './Text'
import { shallow } from 'enzyme'

describe('Text component', () => {

  it('renders correctly with translate', () => {
    const wrapper = shallow(
      <Text
        currentLang={'en'}
      >
        appTitle
      </Text>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly without translate', () => {
    const wrapper = shallow(
      <Text
        currentLang={'en'}
        translate={false}
      >
        Test
      </Text>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
