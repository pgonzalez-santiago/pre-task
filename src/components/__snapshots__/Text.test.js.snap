// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component renders correctly with translate 1`] = `ShallowWrapper {}`;

exports[`Text component renders correctly without translate 1`] = `ShallowWrapper {}`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;
